buildSrc/src/main/groovy/io.deephaven.java-test-conventions.gradle

```diff
@@ -0,0 +1,88 @@
+plugins {
+    id 'java'
+    id 'io.deephaven.java-jacoco-conventions'
+}
+
+def testJar = project.tasks.register 'testJar', Jar, { Jar jar ->
+    jar.from project.sourceSets.test.output
+    jar.classifier = 'test'
+}
+
+configurations {
+    testOutput.extendsFrom testRuntimeClasspath
+}
+
+artifacts {
+    testOutput testJar
+    archives testJar
+}
+
+project.tasks.withType(Test).all { Test t ->
+    t.with {
+        t.defaultCharacterEncoding = 'UTF-8'
+
+        onlyIf { TestTools.shouldRunTests(project) }
+
+        useJUnit()
+
+        new File("$rootDir/tmp/workspace".toString()).mkdirs()
+        new File("$rootDir/tmp/logs".toString()).mkdirs()
+
+        // pass -PforceTest=true to run test again after they've passed once,
+        // and input source has not changed (i.e. invalidate caching)
+        if (findProperty('forceTest') == 'true') {
+            // not gonna stay uptodate very well like this: :-)
+            inputs.property("forceTest", UUID.randomUUID().toString())
```

Comment:
dumb question, but isn't `outputs.upToDateWhen { -> false }` more concise? Either optionally add that lambda, or read forceTest in there, rather than create a garbage value?

Reply:
Sure, happy to adopt that.